base/compiler/inferencestate.jl

```diff
@@ -41,6 +41,10 @@ mutable struct InferenceState
     inferred::Bool
     dont_work_on_me::Bool
 
+    # cached results of calling `_methods_by_ftype`, including `min_valid` and
+    # `max_valid`, to be used in inlining
+    matching_methods_cache::IdDict{DataType, Tuple{Any, UInt, UInt}}
```

Comment:
```suggestion
    matching_methods_cache::IdDict{Any, Tuple{Any, UInt, UInt}}
```
@vtjnash what do you think?